src/asm.rs

```diff
@@ -436,7 +435,7 @@ impl<'a, 'gcc, 'tcx> AsmBuilderMethods<'tcx> for Builder<'a, 'gcc, 'tcx> {
             }
         }
 
-        if !intel_dialect {
+        if att_dialect {
             template_str.push_str(INTEL_SYNTAX_INS);
```

Comment:
Won't this emit `.intel_syntax` on non-x86 targets?

Reply:
```rust
let att_dialect = is_x86 && options.contains(InlineAsmOptions::ATT_SYNTAX);
//                ^^^^^^^^^
```
;)